server/auth_callout.go

```diff
@@ -159,30 +174,60 @@ func (s *Server) processClientOrLeafCallout(c *client, opts *Options) (authorize
 		// Apply to this client.
 		targetAcc := acc
 		// Check if we are being asked to switch accounts.
-		if aname := juc.Audience; aname != _EMPTY_ {
+		// FIXME: this is a directive, shouldn't we be checking here if the
+		//   audience account matches the jwtIssuer, because that is the account...
+		if aname := arc.Audience; aname != _EMPTY_ {
```

Comment:
I think this conditional can go away and the line above can become:

```
targetAcc := s.LookupAccount(jwtIssuer)
```

Reply:
Almost - while on the surface the idea of the authcallout is to create a credential that maps to an account, the code is currently treating audience as a signal to switch accounts.